### PR TITLE
ci: Run Clippy during pull_request event.

### PR DESCRIPTION
**Pull request by duesee**

### CODE REVIEW

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
This shouldn't be touched before there's a good solution for enabling clippy for everyone.

**Comment by duesee** [comment]
As discussed privately, we won't use the Clippy action for now and just run `cargo clippy` directly.

**Review by franziskuskiefer — APPROVED** [review]
thanks

File: .github/workflows/clippy.yml [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
   clippy_check:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
       - run: rustup component add clippy
-      - uses: actions-rs/clippy-check@v1
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          args: -p openmls --tests -- -D warnings
+      - run: cargo clippy -p openmls --tests -- -D warnings
```
Comment (raphaelrobert) [inline .github/workflows/clippy.yml:18]: I think you need to specify something like ` --message-format=json` so that GH can parse it for annotations.

Comment (duesee) [reply .github/workflows/clippy.yml:18]: Oh, good point. I'll make a follow-up.

**Pull request merged**